services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/long_running_tasks.py

```diff
@@ -221,54 +224,80 @@ async def task_restore_state(
     mounted_volumes: MountedVolumes,
     app: FastAPI,
 ) -> None:
-    progress.update(message="checking files", percent=0.0)
-    # first check if there are files (no max concurrency here, these are just quick REST calls)
-    paths_exists: list[bool] = await logged_gather(
-        *(
-            data_manager.exists(
+    # NOTE: the legacy data format was a zip file
+    # this method will maintain retro compatibility.
+    # The legacy archive is always downloaded and decompressed
+    # if found. If the `task_save_state` is successful the legacy
+    # archive will be removed.
+    # When the legacy archive is detected it will have precedence
+    # over the new format.
+    # NOTE: this implies that the legacy format will always be decompressed
+    # until it is not removed.
+
+    project_id_str = f"{settings.DY_SIDECAR_PROJECT_ID}"
+    node_uuid_str = f"{settings.DY_SIDECAR_NODE_ID}"
+
+    async def _restore_state_folder(state_path: Path) -> None:
+        state_archive_exists = await data_manager.state_metadata_entry_exists(
+            user_id=settings.DY_SIDECAR_USER_ID,
+            project_id=project_id_str,
+            node_uuid=node_uuid_str,
+            path=state_path,
+            is_archive=True,
+        )
+        state_directory_exists = await data_manager.state_metadata_entry_exists(
+            user_id=settings.DY_SIDECAR_USER_ID,
+            project_id=project_id_str,
+            node_uuid=node_uuid_str,
+            path=state_path,
+            is_archive=False,
+        )
+        if state_archive_exists:
+            with log_context(_logger, logging.INFO, "restoring legacy data archive"):
+                await data_manager.pull_legacy_archive(
+                    user_id=settings.DY_SIDECAR_USER_ID,
+                    project_id=project_id_str,
+                    node_uuid=node_uuid_str,
+                    destination_path=state_path,
+                    io_log_redirect_cb=functools.partial(
+                        post_sidecar_log_message, app, log_level=logging.INFO
+                    ),
+                    progress_bar=root_progress,
+                )
+        elif state_directory_exists:
```

Comment:
NOTE that `state_directory_exists` does not need to be evaluated if `state_archive_exists == True` while in this code you call both.
Simply:
```python
if state_archive_exists := await data_manager.state_metadata_entry_exists(is_archive=True, **kwargs):
   # ...
elif state_directory_exists : = await data_manager.state_metadata_entry_exists(is_archive=False, **kwargs):
  # ...
```

Reply:
Ahh you are suggesting to move it. Yes that can work.

Reply:
I think we can live with this extra check.

Reply:
i thnk it is more readable but i accept your position as well